### PR TITLE
Fix #219

### LINKED ISSUE #219: Windows 1.37.0 with PR#204 is breaking existing cookbooks using windows_task resource
Unfortunately, Pull Request#204 breaks all of our existing internal cookbooks with the windows_task resource using this new pr as well as chef-client::task...

https://github.com/opscode-cookbooks/chef-client/issues/313

### PR DESCRIPTION
**Pull request by jdmundrawala**
The problem is we try to set password on new_resource to nil.
The correct way to change a value is to use (), and not =. However,
you can't use (nil) because that means return me the current value.

### CODE REVIEW

File: providers/task.rb [hunk 1]
```diff
@@ -241,3 +240,7 @@ def validate_create_day
     end
   end
 end
+
+def password
+  @new_resource.password if !SYSTEM_USERS.include?(@new_resource.user.upcase)
```
Comment (btm) [inline providers/task.rb]: unless instead of "if !"

File: providers/task.rb [hunk 2]
```diff
@@ -241,3 +240,7 @@ def validate_create_day
     end
   end
 end
+
+def password
```
Comment (btm) [inline providers/task.rb]: `#password_unless_user_exists` ?

File: providers/task.rb [hunk 3]
```diff
@@ -41,7 +41,7 @@
     options['ST'] = @new_resource.start_time unless @new_resource.start_time.nil?
     options['TR'] = "\"#{@new_resource.command}\" "
     options['RU'] = @new_resource.user
-    options['RP'] = @new_resource.password if @new_resource.password
+    options['RP'] = password if password
```
Comment (btm) [inline providers/task.rb]: "password if password" sounds like we're setting the password if we have one. it'd be nice to change the name of the method to something that conveys more of its logic.

**Comment by ksubrama** [comment]
:+1: modulo btm's comments.

**Comment by jdmundrawala** [comment]
@btm better

**Comment by btm** [comment]
:+1:

**Pull request merged**